agent/command/generate.go

```diff
@@ -110,13 +110,13 @@ func (c *generateTask) Execute(ctx context.Context, comm client.Communicator, lo
 			if err != nil {
 				return true, err
 			}
-			// only retry if the errors are with saving the project
 			if len(generateStatus.Errors) > 0 {
 				fullErr := errors.New(strings.Join(generateStatus.Errors, ", "))
-				if strings.Contains(fullErr.Error(), evergreen.RetryGenerateTasksError) {
-					return true, fullErr
+				// if the error isn't related to saving the generated task, log it but still retry in case of race condition
```

Comment:
What happens on the server side? I would have thought that a later job might also fail. Does this imply that a second, racing job will succeed?

Reply:
Ah I think it's not that a future racing job will succeed but that an in progress job that has only done some of the modification will succeed.

Reply:
That being said, I feel like we never get stuck in the situation of "this config has been modified but tasks/variants haven't been created" (i.e. if the app server crashed halfway through the job) which maybe implies we _could_ do the slightly bad lock from the previous commit. However that still has a chance of failing, whereas this might retry too many times in some cases but it won't fail incorrectly.

Reply:
On the server side, does returning this error in `j.generate` cause the task to be marked as an error? What happens in that case?

Reply:
yes, although this was always true. We could erase the error from the task if the task is successfully generated in the future.